From 883ecf6d452b55a819440719171969027d324171 Mon Sep 17 00:00:00 2001
From: Bandana <aurelende@protonmail.com>
Date: Fri, 3 Apr 2020 13:24:47 -0400
Subject: [PATCH 05/12] paperdoll fix

Signed-off-by: Bandana <aurelende@protonmail.com>
---
 Intersect.Client/Entities/Entity.cs | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/Intersect.Client/Entities/Entity.cs b/Intersect.Client/Entities/Entity.cs
index 85498404..4db17baa 100644
--- a/Intersect.Client/Entities/Entity.cs
+++ b/Intersect.Client/Entities/Entity.cs
@@ -1140,6 +1140,22 @@ public virtual void DrawEquipment(string filename, int alpha)
                     case 3:
                         d = 2;
 
+                        break;
+                    case 4:
+                        d = 1;
+
+                        break;
+                    case 5:
+                        d = 2;
+
+                        break;
+                    case 6:
+                        d = 1;
+
+                        break;
+                    case 7:
+                        d = 2;
+
                         break;
                 }
 
-- 
2.25.0.windows.1

